semgrep_output_v1.atd

```diff
@@ -914,6 +920,14 @@ type dependency_parser_error = {
   ?text: string option;
 }
 
+(*****************************************************************************)
+(* Semgrep Secrets *)
+(*****************************************************************************)
+
+type historical_info = {
+  git_commit: sha1;
+  git_commit_timestamp: (* ISO 8601 datetime *) string;
```

Comment:
there is a datetime type at the beginning of this file. Would be good to use a consistent type and format for dates.

Reply:
Sure, I can change this and use `string wrap` for the alias I think.

Reply:
It looks like we somewhat use the existing `ATD_string_wrap.Datetime` but only internally for the purpose of conversion to a string in osemgrep output. As-is, that module seems to use git's "ISO 8601-like" format without a timezone. I think it would be better if we used a more standard format (e.g., RFC 3339), and use a dedicated date/time library (we already use `Timedesc` for some other purposes) for handling that.

I've changed `datetime` here to use the wrapper module and plan on updating the code in semgrep to use `Timedesc` in the wrapper module, have it output RFC 3339 formatted datetimes and update the one existing usage of `ATD_string_wrap.Datetime` to no longer use that module but have the same behaviour with the idea being we can update it to be the same in the future. I hesitate to change the format of the existing field along with this unless I can get confirmation that the existing handling of timestamps would allow us to send a timezone (which would seem reasonably likely to me).

Reply:
After some investigation it appears we only ever use https://docs.python.org/3/library/datetime.html#datetime.date.fromisoformat in the app, which should also handle this, so we ought to be covered there.